### PR TITLE
remove/adapt references to yahoo.com

### PR DESCRIPTION
**Pull request by brandelune**

### CODE REVIEW

File: release/changes.txt [hunk 1]
```diff
@@ -8660,6 +8660,7 @@ Bug fix release:
 
 - Fixed Bad Segmentation of some files,
   as documented in: http://groups.yahoo.com/group/OmegaT/message/672
+  (https://sourceforge.net/p/omegat/mailman/message/36811754/)
```
Comment (miurahr) [inline release/changes.txt:8663]: IMHO, yahoo URL can be removed because old URL had already been unreachable.

Comment (brandelune) [reply release/changes.txt:8663]: I preferred to leave it for historical purpose, but you're right.

File: test/src/org/omegat/tokenizer/TokenizerTest.java [hunk 2]
```diff
@@ -188,6 +188,7 @@ public void testChinese() {
      * @see <a href=
      *      "https://groups.yahoo.com/neo/groups/OmegaT/conversations/messages/28395">
      *      User group discussion</a>
+     * @see <a href="https://sourceforge.net/p/omegat/mailman/message/36839317/">Sourceforge archive</a>
```
Comment (miurahr) [inline test/src/org/omegat/tokenizer/TokenizerTest.java:191]: We can remove yahoo URL here, too.

File: release/changes.txt (hunk 1 above)
Comment (amake) [reply release/changes.txt:8663]: Presumably the original link was to a threaded view. The equivalent view now would be https://sourceforge.net/p/omegat/mailman/omegat-users/thread/cflh3i%2Bkbe1%40.../#msg36811754

File: test/src/org/omegat/tokenizer/TokenizerTest.java (hunk 2 above)
Comment (brandelune) [reply test/src/org/omegat/tokenizer/TokenizerTest.java:191]: Sure, but this one I was not 100% sure that it was pointing at the right mail.

File: release/changes.txt (hunk 1 above)
Comment (miurahr) [reply release/changes.txt:8663]: >  for historical purpose,

IMHO, historical URL can be seen in git history when necessary.

File: test/src/org/omegat/tokenizer/TokenizerTest.java (hunk 2 above)
Comment (brandelune) [reply test/src/org/omegat/tokenizer/TokenizerTest.java:191]: You're right. I just did not feel comfortable removing information. If you think it is not relevant, I am totally fine with your decision.

**Comment by amake** [comment]
Thanks. I've applied to `master`.

**Pull request closed without merging**